charms/istio-pilot/config.yaml

```
@@ -3,3 +3,7 @@ options:
     default: istio-gateway
     description: Comma-separated list of gateways to create by default
     type: string
+  namespaced-custom-gateways:
+    default: ""
+    description: Comma-separated list of namespaced custom gateways to use
+    type: string
```

I'm not sure that it's useful to include "namespaced" in the option name and it makes it a bit unwieldy.  Suggestion:
```suggestion
  custom-gateways:
    default: ""
    description: >-
      Comma-separated list of custom or manually configured Gateway resources to use.
      Names can include the namespace, in the form "namespace/gateway", or defaults to
      the model in which the charm is deployed.
    type: string
```

(Note: The default for the namespace is [handled automatically by Istio](https://istio.io/latest/docs/reference/config/networking/virtual-service/#VirtualService-gateways), and the explicit namespace in the current template is actually superfluous.)

---

Updated.